spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/FilteringBatchAcknowledgingMessageListenerAdapter.java

```diff
@@ -65,7 +65,7 @@ public FilteringBatchAcknowledgingMessageListenerAdapter(BatchAcknowledgingMessa
 	 */
 	public FilteringBatchAcknowledgingMessageListenerAdapter(BatchAcknowledgingMessageListener<K, V> delegate,
 			RecordFilterStrategy<K, V> recordFilterStrategy, boolean ackDiscarded) {
-		super(recordFilterStrategy);
+		super(delegate, recordFilterStrategy);
 		this.delegate = delegate;
```

Comment:
Maybe we can just store `delegate` in `super`?
Right, minor, but I don't know already what to complain about here.
Now I understand all that thread callback dance and just enjoy watching to the code :smile: